Added new setting to enable Recommendations

sagzy opened the pull request:
closes https://github.com/TryGhost/Product/issues/3772

- Recommendations are disabled by default and will get enabled when a first recommendation is added. The "recommendations_enabled" setting will be not shown explicitly in Admin, it will rather be set implicitly based on whether there is at least one recommendation
- The "recommendations_enabled" setting belongs to a new setting group called "recommendations"

[Tech Spec](https://www.notion.so/ghost/Recommentions-Tech-Spec-598f09e4109f4eef8d9e32db7245a90c?pvs=4)

sagzy committed:
Added new setting to enable Recommendations

refs https://github.com/TryGhost/Product/issues/3772

- "recommendations_enabled" setting allows to enable/disable the Recommendations feature in Admin
- "recommendations_enabled" setting belongs to a new setting group called "recommendations"
- "recommendations_enabled" is set to true by default (i.e., Recommendations are enabled by default)

SimonBackx commented on ghost/core/core/server/data/schema/default-settings/default-settings.json:
```diff
@@ -565,5 +565,15 @@
+    },
+    "recommendations": {
+        "recommendations_enabled": {
+            "defaultValue": "true",
```
```suggestion
            "defaultValue": "false",
```

sagzy replied:
Updated, following the Slack discussion [here](https://ghost.slack.com/archives/C04TMVA1D7A/p1693310637853019)

SimonBackx commented on ghost/core/core/server/data/migrations/versions/5.61/2023-08-29-09-32-32-add-recommendations-enabled-setting.js:
```diff
@@ -0,0 +1,8 @@
+
+module.exports = addSetting({
+    key: 'recommendations_enabled',
+    value: 'true',
```
```suggestion
    value: 'false',
```

sagzy replied:
Updated, following the Slack discussion [here](https://ghost.slack.com/archives/C04TMVA1D7A/p1693310637853019)

SimonBackx commented:
We probably also want to update `ghost/core/test/utils/fixtures/default-settings.json`, `ghost/core/test/utils/fixtures/default-settings-browser.json` and `ghost/core/core/shared/settings-cache/public.js`

sagzy committed:
Merge remote-tracking branch 'origin/main' into dml-migration/setting-to-enable-recommendations

sagzy committed:
Recommendations are now disabled by default

- also added "recommendations_enabled" setting into fixtures and the settings cache

sagzy commented:
> We probably also want to update `ghost/core/test/utils/fixtures/default-settings.json`, `ghost/core/test/utils/fixtures/default-settings-browser.json` and `ghost/core/core/shared/settings-cache/public.js`

Thanks! Added these as well

sagzy committed:
Merge branch 'main' into dml-migration/setting-to-enable-recommendations

sagzy committed:
Fixed e2e tests

sagzy committed:
Merge branch 'main' into dml-migration/setting-to-enable-recommendations

sagzy committed:
Fixed content setting e2e tests

sagzy committed:
Fixed e2e version tests
